### PR TITLE
[feature] create a database from another one

### PR DESCRIPTION
**Pull request by yuriolive**
Adding the feature to create a database from another one, using the clone command in Snowflake. Similar to `from_share` feature.

https://docs.snowflake.net/manuals/sql-reference/sql/create-clone.html

### CODE REVIEW

File: pkg/resources/database.go [hunk 1]
```diff
@@ -34,6 +34,12 @@ var databaseSchema = map[string]*schema.Schema{
 		Optional:    true,
 		ForceNew:    true,
```
Comment (edulop91) [inline pkg/resources/database.go]:
```suggestion
		ForceNew:    true,
		ConflictsWith: []string{"from_database},
```

File: pkg/resources/database.go [hunk 2]
```diff
@@ -55,10 +61,18 @@ func Database() *schema.Resource {
 
 // CreateDatabase implements schema.CreateFunc
 func CreateDatabase(data *schema.ResourceData, meta interface{}) error {
-	_, ok := data.GetOk("from_share")
-	if ok {
-		return createDatabaseFromShare(data, meta)
+	_, okShare := data.GetOk("from_share")
+	_, okDatabase := data.GetOk("from_database")
```
Comment (edulop91) [inline pkg/resources/database.go]:
```suggestion
	_, ok = data.GetOk("from_database")
	if ok {
	  return createDatabaseFromDatabase(data, meta)
	}
```

File: pkg/resources/database.go [hunk 3]
```diff
@@ -55,10 +61,18 @@ func Database() *schema.Resource {
 
 // CreateDatabase implements schema.CreateFunc
 func CreateDatabase(data *schema.ResourceData, meta interface{}) error {
-	_, ok := data.GetOk("from_share")
-	if ok {
-		return createDatabaseFromShare(data, meta)
+	_, okShare := data.GetOk("from_share")
```
[the remark this answers is not in the document]
Comment (edulop91) [reply pkg/resources/database.go]: We can simplify this logic since terraform takes care of collisions for us.

**Comment by yuriolive** [comment]
@edulop91 Changed the code to use ConflictWith and the others nit-picks. I'm trying to discover why is failing in the CI, I'm getting in the database acceptance test: `Missing required argument: The argument "username" is required, but was not set.` Any idea?

**Comment by edulop91** [comment]
Thank you! TravisCI restricts outside collaborators from running jobs with protected vars. I’ll run them shortly.

**Comment by edulop91** [comment]
Confirming tests pass

**Pull request merged**